File: PruebaPicasso/app/src/main/java/es/iesnervion/avazquez/pruebapicasso/fragments/BreedListFragment.java
package es.iesnervion.avazquez.pruebapicasso.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProviders;

import java.util.ArrayList;

import es.iesnervion.avazquez.pruebapicasso.R;
import es.iesnervion.avazquez.pruebapicasso.viewModel.SharedViewModel;

public class BreedListFragment extends Fragment implements AdapterView.OnItemClickListener {

    SharedViewModel viewModel;
    ListView listView;
    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_breed_list, container, false);
        ArrayList<String> razasPerro = new ArrayList<>();
        razasPerro.add("akita");
        razasPerro.add("malamute");
        razasPerro.add("husky");
        razasPerro.add("labrador");
        viewModel = ViewModelProviders.of(getActivity()).get(SharedViewModel.class);
        listView = view.findViewById(R.id.listViewRazas);
        ArrayAdapter<String> adapter =
                    new ArrayAdapter<>(getContext(), android.R.layout.simple_list_item_1, razasPerro);
        listView.setAdapter(adapter);

        listView.setOnItemClickListener(this);


        return view;
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
        /*Cuando un elemento se clicka, pongo ese elemento
        * como raza seleccionada en el VM*/
        String razaElegida = "";
        //Cojo la raza elegida
        if(position >= 0){
            razaElegida = (String) listView.getItemAtPosition(position);
        }


        //Le digo al view model que este es el contacto seleccionado
        viewModel.setRazaElegida(razaElegida);
    }
}

File: EjemploKotlinRetrofit2/app/src/main/java/es/iesnervion/avazquez/ejemplokotlinretrofit2/fragments/BreedListFragment.kt
package es.iesnervion.avazquez.ejemplokotlinretrofit2.fragments

import android.content.Context
import android.net.Uri
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.ListView
import androidx.lifecycle.ViewModelProviders

import es.iesnervion.avazquez.ejemplokotlinretrofit2.R
import es.iesnervion.avazquez.ejemplokotlinretrofit2.viewModel.SharedViewModel

class BreedListFragment : Fragment() {

    lateinit var viewModel : SharedViewModel
    lateinit var listView: ListView

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        var view: View = inflater.inflate(R.layout.fragment_breed_list, container, false)
        viewModel = ViewModelProviders.of(activity!!).get(SharedViewModel::class.java)
        listView = view.findViewById(R.id.listViewRazas)
        val listaRazas = arrayListOf<String>("akita","husky","labrador","malamute")
        var adapter = activity?.let {
            ArrayAdapter(
                it,
                android.R.layout.simple_list_item_1,
                listaRazas)
        }
        listView.adapter = adapter
        listView.setOnItemClickListener{ parent, view, position, id ->
            //Cuando se pulse una raza, se actualizará en el view model
            viewModel.setRazaElegida(listView.getItemAtPosition(position) as String)
        }


        return view
    }



}
